classes/Member.php

```diff
@@ -142,6 +142,48 @@ public function image() {
 
     }
 
+    // TODO: some of this probably wants to go elsewhere
+    public function getMostRecentMembership() {
+        $last_cons = '';
+        $last_house = null;
+        $last_party = null;
+        if ( $this->current_member_anywhere() ) {
+            $houses = array_keys(array_filter($this->current_member(), 'strlen'));
+            $last_cons = $this->constituency;
+            $last_party = $this->party;
+            $last_house = min($houses);
+        } else {
+            $max_date = null;
+            foreach ( array_keys($this->left_house) as $house ) {
+                if ($this->left_house[$house]['date'] > $max_date ) {
+                    $max_date = $this->left_house[$house]['date'];
+                    $last_cons = $this->left_house[$house]['constituency'];
+                    $last_party = $this->left_house[$house]['party'];
+                    $last_house = $house;
+                }
+            }
+        }
+        $details = array(
+            'entered_house' => '',
+            'left_house' => '',
+            'cons' => $last_cons,
+            'party' => $last_party,
+            'house' => $last_house,
+            'rep_name' => $this->getRepNameForHouse($last_house)
+        );
+
+        $entered_house = $this->entered_house($last_house);
+        $left_house = $this->left_house($last_house);
+        if ( isset($entered_house['date']) ) {
```

Comment:
If `$last_house` is an actual house the member was in, then I would have thought these dates would have to be set.